firedrake/fml/form_manipulation_language.py

```diff
@@ -280,27 +304,31 @@ def __truediv__(self, other):
 
     __rmul__ = __mul__
 
-    def __iter__(self):
+    def __iter__(self) -> Sequence:
         """Returns an iterable of the terms in the labelled form."""
         return iter(self.terms)
 
-    def __len__(self):
+    def __len__(self) -> int:
         """Returns the number of terms in the labelled form."""
         return len(self.terms)
 
-    def label_map(self, term_filter, map_if_true=identity,
-                  map_if_false=identity):
+    def label_map(
+        self,
+        term_filter: Callable,
```

Comment:
```
term_filter: Callable[[Term], bool]
```

and `map_if_true: Callable[[Term], Optional[Term]` and similarly for `map_if_false`.